### DISCUSSION TITLE
Support running multiple worlds at the same time?

### DISCUSSION THREAD

**Discussion by Peabo83** [Nimdy/Dedicated_Valheim_Server_Script#227] [General]
This script is great, I setup a ubuntu valheim server by hand and it takes much longer. Thanks for writing this repo!

A friend asked if I could host his world and I was wondering if there was some way I could host 2 worlds on the same box while using this script.

**Comment by LordDumoss** [comment]
I agree.

Functionality "Add Valheim server +1"
And use "almost" all the same code as the install 

Just add a prompt for a port request (making sure it was not in current use somehow).

Then script does work leaving ... something like

start_valheim_<world>.sh
/etc/systemd/system/valheimserver<world>.service

Currently, I am coping the 2 files created by this script and making the changes needed.

**Reply by LordDumoss** [reply]
Need to fix ...

start_valheim_world.sh
/etc/systemd/system/valheimserver_world.service

Would need the start and stop service script to handle "world" name and original scripts 

start_valheim_default.sh
/etc/systemd/system/valheimserver_default.service

**Comment by opensolutionsgroup** [comment]
Using the beta version with the new language file and was able to create 2 new servers on same box.  Had to migrate by hand the .db and .flw from old to a new location based on where the new script expected files to be.  Also had to clean up the old file locations that are no longer used.

Only glitch I had was that '-' and '_' are no supported in passwords or world names.  I had to use https://geekstrom.de/valheim/fwl/ to rename my world file that contained a '-'.   

I don't use a firewall on this box, so I didn't care that this wasn't working yet. 

Great work, thank-you.

**Reply by Nimdy** [reply]
thanks for testing, we are trying. almost there
